src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -254,11 +248,56 @@ private void TraceAndSendExceptionNotification(string exceptionDetails)
             }
         }
 
+        private OrchestrationInvocationResult GetInvocationResult(object ooprocResult, Exception ex = null)
+        {
+            JObject resultJObject = this.ParseOOProcResult(ooprocResult);
+            string resultJSONString = resultJObject.ToString();

```

Comment:
Sometimes the OOProc result is already a string, right? Could we optimize this further by having `ParseOOProcResult` return both the string and JObject representation, and doing only the minimum amount of conversion necessary?